examples/lint: Remove unused `Result::map`

CohenArthur opened the pull request:
`with_lints` already returns a `Result<(), _>`, so mapping the `Ok` value
to `()` does nothing.

GuillaumeGomez commented on Cargo.lock:
```diff
@@ -92,7 +92,7 @@ dependencies = [
 [[package]]
 name = "rustc-tools"
-version = "0.2.0"
+version = "0.3.0"
```
Ah forgot to build after updating the version. ^^'

GuillaumeGomez commented:
Thanks!

CohenArthur committed:
examples/lint: Remove unused `Result::map`

`with_lints` already returns a `Result<(), _>`, so mapping the `Ok` value
to `()` does nothing.
